openmdao/docs/features/model_visualization/n2_details.rst

```diff
@@ -0,0 +1,297 @@
+
+
+.. raw:: html
+    :file: examples/n2_circuit_with_unconnected_input.html
+
+
+Model Hierarchy
+---------------
+
+The model hierarchy is on the left side of the diagram. It is actually a mix of Systems and variables, both
+inputs and outputs.
+
+The top System, the root of the model, is on the left and the subsystem hierarchy is displayed to right.
+
+The colors indicate the type of
+item in the hierarchy. To see what the colors represent, an optional legend can be displayed using the |show_legend| button.
+Here is the part of the legend that explains the colors of the model hierarchy.
+
+.. image::
+    images/systems_and_variables_legend.png
+
+You can click in the model hierarchy to collapse and uncollapse sections of the hierarchy. Left clicking on a
+subsystem zooms the hierarchy display to just that subsystem in the hierarchy. Right clicking on a subsystem collapses all of the
+hierarchy to the right of it.
+
+Connection Matrix
+-----------------
+The matrix in the middle of the diagram shows what items in the model are connected.
+
+The color of an item on a diagonal cell matches the color of the rightmost element in model hierarchy. The colors
+for the off-diagonal items, which indicate connections, are always black.
+
+The symbol of an item in the connection matrix indicates whether the variable is a scalar, vector, or collapsed variable.
+
+.. image::
+    images/connection_matrix_legend.png
+
+Black boxes outline items in the matrix to show the grouping of the variables of the lowest level components.
+
+When you hover over a connection in the matrix, arrows are drawn to indicate connections. The behavior is different
+depending on whether you hover over a diagonal or an off-diagonal cell.
+
+When hovering over a diagonal item, the explicit connections to/from that variable are displayed. Red arrows indicate
+that element is the target of the connection. A green arrow indicates it is the source.
+
+When hovering over a off-diagonal, the behavior differs depending on whether the cell is in the lower or upper
+triangular part of the diagram.
+
+In the lower triangular, the marked cells represent upstream (in the sense of execution order of the
+components) connections. All the connections involved in that cycle
+are drawn to indicate which systems and components are involved in the convergence loop. Downstream connection arrows
+are green and upstream connections are red.
+
+In the upper triangular, the marked cells represent downstream connections. The red arrow that is drawn shows
+that connection.
+
+Clicking in the matrix on a connection or diagonal, lets us pin connections arrows so that they don't
+disappear when you hover away from the cell. You can clear these connections arrows by using the Clear Arrows and
+Connections button |clear_conn_arrows| in the toolbar.
+
+Solver Structure
+----------------
+Each System in a model can potentially have a linear and non-linear model associated with it. On the right side
```

Comment:
nonlinear doesn't need to be hyphenated

Reply:
thanks there in other places in the docs